src/test/java/ca/utoronto/fitbook/unit/UpvotePostsServiceTest.java

```diff
@@ -0,0 +1,70 @@
+package ca.utoronto.fitbook.unit;
+
+import ca.utoronto.fitbook.BaseTest;
+import ca.utoronto.fitbook.adapter.persistence.firebase.UserFirebaseRepository;
+import ca.utoronto.fitbook.adapter.persistence.localmemory.PostLocalMemoryRepository;
+import ca.utoronto.fitbook.adapter.persistence.localmemory.UserLocalMemoryRepository;
+import ca.utoronto.fitbook.application.port.in.UpvotePostsUsecase;
+import ca.utoronto.fitbook.application.port.in.command.UpvotePostsCommand;
+import ca.utoronto.fitbook.application.service.UpvotePostsService;
+import ca.utoronto.fitbook.entity.User;
+import org.junit.jupiter.api.*;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import java.util.*;
+
+public class UpvotePostsServiceTest extends BaseTest {
+
+    @Autowired
+    private UserLocalMemoryRepository userLocalMemoryRepository;
+    private PostLocalMemoryRepository postLocalMemoryRepository;
+    private UpvotePostsUsecase upvotePostsUsecase;
+    private User testUser1;
+    private User testUser2;
+
+    @BeforeAll
+    public void init() {
+        this.userLocalMemoryRepository = new UserLocalMemoryRepository();
+        this.postLocalMemoryRepository = new PostLocalMemoryRepository();
+        this.upvotePostsUsecase = new UpvotePostsService(postLocalMemoryRepository, userLocalMemoryRepository);
+
+        testUser1 = TestUtilities.randomUser();
+        testUser2 = TestUtilities.randomUser();
+
+        userLocalMemoryRepository.save(testUser1);
+        userLocalMemoryRepository.save(testUser2);
+    }
+
+    @AfterAll
+    public void cleanUp() {
+        userLocalMemoryRepository.delete(testUser1.getId());
+        userLocalMemoryRepository.delete(testUser2.getId());
+    }
+    @AfterEach
+    public void resetLikes(){
+        testUser1.setLikedPostIdList(new ArrayList<>());
+        testUser2.setPostIdList(new ArrayList<>());
+        userLocalMemoryRepository.save(testUser1);
+        userLocalMemoryRepository.save(testUser2);
+    }
+
```

Comment:
There should be a test for the like functionality working, and then checking that the database version of each entity was also updated accordingly.